pkg/brownfield/pathmaps.go

```diff
@@ -67,6 +75,60 @@ func MergePathMaps(pathMapBuckets ...[]n.ApplicationGatewayURLPathMap) []n.Appli
 	return merged
 }
 
+// MergePathMapsWithBasicRule merges a Url pathmap with a basic routing rule
+func MergePathMapsWithBasicRule(pathMap *n.ApplicationGatewayURLPathMap, rule *n.ApplicationGatewayRequestRoutingRule) *n.ApplicationGatewayURLPathMap {
+	// Replace the default values in the path map with values from the routing rule
+	pathMap.DefaultBackendAddressPool = rule.BackendAddressPool
+	pathMap.DefaultBackendHTTPSettings = rule.BackendHTTPSettings
+	pathMap.DefaultRedirectConfiguration = rule.RedirectConfiguration
+	pathMap.DefaultRewriteRuleSet = rule.RewriteRuleSet
+	return pathMap
+}
+
+// MergePathRules merges list of lists of pathMaps into a single list, maintaining uniqueness.
+func MergePathRules(pathRulesBucket ...*[]n.ApplicationGatewayPathRule) *[]n.ApplicationGatewayPathRule {
+	uniq := make(pathRulesByName)
+	for _, bucket := range pathRulesBucket {
+		for _, pathRule := range *bucket {
+			uniq[pathRuleName(*pathRule.Name)] = pathRule
+		}
+	}
+	var merged []n.ApplicationGatewayPathRule
+	for _, pathRule := range uniq {
+		merged = append(merged, pathRule)
+	}
+	return &merged
+}
+
+// LookupPathMap using resourceID in the list of path map
+func LookupPathMap(pathMaps *[]n.ApplicationGatewayURLPathMap, resourceID *string) *n.ApplicationGatewayURLPathMap {
+	for idx, pathMap := range *pathMaps {
+		if *pathMap.ID == *resourceID {
+			return &(*pathMaps)[idx]
+		}
+	}
+
+	return nil
+}
+
+// DeletePathMap deletes a path map from the list of path maps and return the new list
+func DeletePathMap(pathMapsPtr *[]n.ApplicationGatewayURLPathMap, resourceID *string) *[]n.ApplicationGatewayURLPathMap {
```

Comment:
This is used in the same package - you could lowercase it and remove the docstring (since it is fairly obvious what it does)